lite/dynamic_verifier.go

```diff
@@ -266,6 +272,57 @@ FOR_LOOP:
 	}
 }
 
+// fetchAndVerifyToHeight will fetch, verify, and store all intermediate
+// full commits from the last trusted one up to the given height h.
+//
+// If all of the above steps are successful it returns the full commit at
+// height h and no error.
+//
+// It returns ErrCommitNotFound if source provider doesn't have the commit for h.
+func (dv *DynamicVerifier) fetchAndVerifyToHeight(h int64) (FullCommit, error) {
+	// Fetch latest full commit from trusted.
+	trustedFC, err := dv.trusted.LatestFullCommit(dv.chainID, 1, h)
+	if err != nil {
+		return FullCommit{}, err
+	}
+	// We already were at the requested height.
+	if trustedFC.Height() == h {
```

Comment:
what if trustedFC.Height is greater than h?

Reply:
This would mean `trustedFC, err := dv.trusted.LatestFullCommit(dv.chainID, 1, h)` is not implemented correctly as it should return the latest full commit with height 1 <= height <= h. As this should not happen, it would be OK to panic in this case.